fastFM/ffm2.pyx

```diff
@@ -77,11 +78,35 @@ def ffm_predict(np.ndarray[np.float64_t, ndim = 1] w_0,
 
     return y
 
+# The main piece of the glue between Python, Cython and C++
+# In essence it wraps the python function so it can be used in C++ space.
+# Convert the python function from a pointer back into a python object and invoke
+# with other parameters. (For now just one, `current_iter`)
+# TODO: Handle exceptions in a better manner
+cdef bool fit_callback_wrapper(string json_str, void* python_function):
+    f = (<object>python_function)
+
+    try:
+        kwargs = json.loads(json_str.c_str())
+
+        print("Arguments from C++: " + str(kwargs))
+    except json.JSONDecodeError as e: # should never happen but just in case
```

Comment:
Okay, now I understand better what you mean. I'm still not to happy with the use of exceptions. We could raise a warning instead and not sanitize the input. We can deal with it on the cpp side.